src/main/java/dev/openfeature/sdk/ProviderAccessor.java

```diff
@@ -0,0 +1,9 @@
+package dev.openfeature.sdk;
+
+/**
+ * Provides access to the future provider for the domain of the client.
+ */
+@FunctionalInterface
+interface ProviderAccessor {
+    FeatureProviderStateManager getProviderStateManager();
```

Comment:
I know we talked that the FeatureProviderStateManager should at best be invisible outside the PrviderRepository, but I don't see another way to make [this](https://github.com/open-feature/java-sdk/pull/1096/files#diff-e0fc1e23041e08fa8dd72864acce5b949f84987f405accd4935f233de281ef14R125) thread safe